### PR TITLE
WIP - new indexing model - updating unit tests

### PR DESCRIPTION
**Pull request by shawnce**
@gribozavr

### CODE REVIEW

**Comment by shawnce** [comment]
The changes to CheckCollectionType.swift have been done by eyeball and search/replace since what it depends on doesn't yet compile.

File: stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift [hunk 1]
```diff
@@ -503,7 +512,7 @@ self.test("\(testNamePrefix).index(of:)/semantics") {
       Optional<CollectionWithEquatableElement.Index>.self,
       &result)
     let zeroBasedIndex = result.map {
-      numericCast(c.startIndex.distance(to: $0)) as Int
+      numericCast(c.distance(from: startIndex, to: $0)) as Int
```
Comment (gribozavr) [inline stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift]: `from: c.startIndex`

File: stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift [hunk 2]
```diff
@@ -523,7 +532,7 @@ self.test("\(testNamePrefix).index(where:)/semantics") {
       return extractValueFromEquatable(candidate).value == test.element.value
     }
     let zeroBasedIndex = result.map {
-      numericCast(c.startIndex.distance(to: $0)) as Int
+      numericCast(c.distance(from: startIndex, to: $0)) as Int
```
Comment (gribozavr) [inline stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift]: `from: c.startIndex`

File: stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift [hunk 3]
```diff
@@ -646,7 +655,7 @@ self.test("\(testNamePrefix).split/semantics") {
 self.test("\(testNamePrefix).prefix(through:)/semantics") {
   for test in prefixThroughTests {
     let c = makeWrappedCollection(test.collection.map(OpaqueValue.init))
-    let index = c.startIndex.advanced(by: numericCast(test.position))
+    let index = c.advance(startIndex, by: numericCast(test.position))
```
Comment (gribozavr) [inline stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift]: `c.startIndex`

File: stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift [hunk 4]
```diff
@@ -660,7 +669,7 @@ self.test("\(testNamePrefix).prefix(through:)/semantics") {
 self.test("\(testNamePrefix).prefix(upTo:)/semantics") {
   for test in prefixUpToTests {
     let c = makeWrappedCollection(test.collection.map(OpaqueValue.init))
-    let index = c.startIndex.advanced(by: numericCast(test.end))
+    let index = c.advance(startIndex, by: numericCast(test.end))
```
Comment (gribozavr) [inline stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift]: `c.startIndex`

File: stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift [hunk 5]
```diff
@@ -674,7 +683,7 @@ self.test("\(testNamePrefix).prefix(upTo:)/semantics") {
 self.test("\(testNamePrefix).suffix(from:)/semantics") {
   for test in suffixFromTests {
     let c = makeWrappedCollection(test.collection.map(OpaqueValue.init))
-    let index = c.startIndex.advanced(by: numericCast(test.start))
+    let index = c.advance(startIndex, by: numericCast(test.start))
```
Comment (gribozavr) [inline stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift]: `c.startIndex`

**Comment by shawnce** [comment]
Fixing

**Comment by shawnce** [comment]
Think things are corrected

**Comment by shawnce** [comment]
Gonna pull back my changes to StdlibUnittest.swift.gyb ... good amount wrong and likely not worth it

**Pull request merged**